### PR TITLE
[INLONG-603]Codec for Go sdk

### PR DESCRIPTION
**Pull request by TszKitLo40**

### CODE REVIEW

**Comment by gosonzhang** [comment]
@charlely, I don’t know Go well, please review it.

From my personal point of view, it is OK

Thanks!

File: tubemq-client-twins/tubemq-client-go/codec/codec.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	// Encode encodes the RPCRequest to bytes.
+	Encode() ([]byte, error)
+}
+
+// RPCResponse represents the RPC response from TubeMQ.
+type RPCResponse interface {
+	// GetSerialNo returns the `serialNo` of the corresponding request.
+	GetSerialNo() uint32
+	// Decode decodes the Response.
+	Decode(response Response) error
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/codec/codec.go:50]: Change Decode to Marshal

File: tubemq-client-twins/tubemq-client-go/codec/codec.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	// Decode will decode the response to frame head and body.
+	Decode() (Response, error)
+}
+
+// RPCRequest represents the RPC request protocol.
+type RPCRequest interface {
+	// GetSerialNo returns the `serialNo` of the corresponding request.
+	GetSerialNo() uint32
+	// Encode encodes the RPCRequest to bytes.
+	Encode() ([]byte, error)
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/codec/codec.go:42]: Change Encode to Unmarshal

File: tubemq-client-twins/tubemq-client-go/codec/codec.go (hunk 1 above)
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/codec/codec.go:50]: Input param is []byte

File: tubemq-client-twins/tubemq-client-go/codec/codec.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+// RPCResponse represents the RPC response from TubeMQ.
+type RPCResponse interface {
+	// GetSerialNo returns the `serialNo` of the corresponding request.
+	GetSerialNo() uint32
+	// Decode decodes the Response.
+	Decode(response Response) error
+	// GetException returns the exception msg.
+	GetException() string
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/codec/codec.go:52]: GetExeption cann't use to common function.

File: tubemq-client-twins/tubemq-client-go/codec/codec.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	// GetStackTrace returns the stack track msg.
+	GetStackTrace() string
```
Comment (charlely) [inline tubemq-client-twins/tubemq-client-go/codec/codec.go:54]: GetStackTrace cann't use to common function.

**Review by charlely — APPROVED** [review]
LGTM

**Pull request merged**